src/libraries/System.Runtime.Loader/tests/ApplyUpdateTest.cs

```diff
@@ -82,5 +82,43 @@ void ClassWithCustomAttributes()
                 Assert.Equal(attrType, cattrs[0].GetType());
             });
         }
+
+        class NonRuntimeAssembly : Assembly
+        {
+        }
+
+        [Fact]
+        public static void ApplyUpdateInvalidParameters()
+        {
+            // Dummy delta arrays
+            var metadataDelta = new byte[20];
+            var ilDelta = new byte[20];
+
+            // Assembly can't be null
+            Assert.Throws<ArgumentNullException>("assembly", () =>
+                MetadataUpdater.ApplyUpdate(null, new ReadOnlySpan<byte>(metadataDelta), new ReadOnlySpan<byte>(ilDelta), ReadOnlySpan<byte>.Empty));
+
+            // Tests fail on non-runtime assemblies
+            Assert.Throws<ArgumentException>(() =>
+                MetadataUpdater.ApplyUpdate(new NonRuntimeAssembly(), new ReadOnlySpan<byte>(metadataDelta), new ReadOnlySpan<byte>(ilDelta), ReadOnlySpan<byte>.Empty));
+
+            // Tests that this assembly isn't not editable
+            Assert.Throws<InvalidOperationException>(() =>
+                MetadataUpdater.ApplyUpdate(typeof(AssemblyExtensions).Assembly, new ReadOnlySpan<byte>(metadataDelta), new ReadOnlySpan<byte>(ilDelta), ReadOnlySpan<byte>.Empty));
+        }
+
+        [Fact]
+        public static void GetCapabilities()
+        {
+            string result = MetadataUpdater.GetCapabilities();
+            Assert.NotNull(result);
+        }
+
+        [Fact]
+        public static void IsSupported()
+        {
+            bool result = MetadataUpdater.IsSupported;
+            Assert.False(result);
+        }
```

Comment:
I think this test will fail on WebAssembly.  The .csproj sets `DOTNET_MODIFIABLE_ASSEMBLIES=debug` at launch

Reply:
What do you suggest for mono? Does the below CI tests include a WebAssembly leg?  It is kind of a silly test maybe I should remove it or skip it on mono.

Reply:
I think we should skip it on Mono.

I'm not 100% clear on why it didn't fail on the CI leg.  Investigating.  I don't think it's worth holding up the whole PR.